### PR TITLE
[AMD][Perf] Add dense-MHA one-shot prefill support paths on gfx950 for DSA models

### PR DESCRIPTION
**Pull request by amd-pedghazi**
## Motivation

DSA models (DeepSeek-V3.2, GLM-5.1/5.2) use a **sparse** MLA prefill (lightning indexer → top-k `index_topk`=2048 → sparse MLA). The backend already has a faster **dense one-shot MHA** prefill (`_forward_standard_mha`) gated by `SGLANG_DSA_PREFILL_DENSE_ATTN_KV_LEN_THRESHOLD`, but it was hard-gated to NVIDIA SM90/SM100, so AMD gfx950 (MI350/MI355) always fell back to sparse.

This PR lands the **gfx950 support paths** needed to make that dense one-shot MHA prefill correct on the HIP fp8 KV layout. It is intentionally split from the small hardware-gate flip so each piece is easy to review.

## Modifications

- **`dsa/dequant_k_cache.py`** — add a `dim_quant == 576` branch for the HIP raw fp8 MLA latent (`kv_lora_rank + qk_rope_head_dim` = 512 + 64, no per-tile scales), distinct from the CUDA 656B layout. Paged gather + fp8→bf16 upcast so prefix-continuation can reload KV for dense MHA.
- **`models/deepseek_common/attention_forward_methods/forward_mha.py`** — on `MHA_ONE_SHOT` full-KV reload, the pre-quantized `kv_a_quanted` only covers the current extend tokens and is stale; route the reloaded bf16 `kv_a` through `kv_b_proj` (internal activation quant).
- **`dsa/dsa_indexer.py`** — extend the skip-logits fast path to also trigger on `not return_indices` (set by the model when the prefill runs dense MHA), skipping the discarded indexer GEMM + top-k at any context. The indexer K-cache is still written, so a later sparse decode step stays correct.
- **`test/registered/amd/test_dsa_dense_mha_gfx950.py`** — gfx950 CI coverage for the dense-fallback correctness cases (reuses the CUDA suite's dense-fallback case maker/runner). Verified on MI350X: 1 passed, 8 subtests.

**Dependency:** none at runtime. Pairs with the gate PR to take effect.

## Checklist
- [x] Format code with pre-commit
- [x] Add unit tests (gfx950 dense-fallback CI coverage)
- [x] Verified on MI350X

---
### CI States

Latest PR Test (Base): :x: [Run #29231673196](https://github.com/sgl-project/sglang/actions/runs/29231673196)
Latest PR Test (Extra): :x: [Run #29231673068](https://github.com/sgl-project/sglang/actions/runs/29231673068)

### CODE REVIEW

File: python/sglang/srt/layers/attention/dsa/dequant_k_cache.py [hunk 1]
```diff
@@ -165,6 +165,62 @@ def _dequantize_k_cache_fast_kernel(
...
+    # quant_ptr is an FP8-typed view; the HIP DSA MLA latent is stored as a plain
+    # bf16->fp8 cast with NO per-tile scales (see memory_pool.set_mla_kv_buffer
+    # HIP branch / mla_buffer.set_mla_kv_buffer_fp8_quant_kernel), so dequant is a
+    # paged gather + fp8->bf16 upcast (scale == 1.0).
+    src = tl.load(quant_ptr + src_token * quant_stride_0 + offs, mask=mask, other=0.0)
+    tl.store(
+        output_ptr + token_id * output_stride_0 + offs,
+        src.to(output_ptr.dtype.element_ty),
+        mask=mask,
+    )
```
[the remark this answers is not in the document]
Comment (amd-pedghazi) [reply python/sglang/srt/layers/attention/dsa/dequant_k_cache.py]: > medium
> 
> On ROCm/HIP platforms (especially gfx950), direct casting from FP8 types (like `float8_e4m3fn` or `float8_e4m3fnuz`) to `bfloat16` in Triton can sometimes be unsupported or generate inefficient/incorrect code depending on the compiler version. It is safer and more portable to cast the loaded FP8 values to `tl.float32` first, and then cast to the output's element type, matching the pattern used in other dequantization kernels in this file.

This has been addressed.
